devices/YarpRobotLoggerDevice/include/BipedalLocomotion/YarpRobotLoggerDevice.h

```diff
@@ -167,10 +174,26 @@ class YarpRobotLoggerDevice : public yarp::dev::DeviceDriver,
     std::vector<std::string> m_codeStatusCmdPrefixes;
 
     std::mutex m_bufferManagerMutex;
+    std::mutex m_textLoggingPortMutex;
     robometry::BufferManager m_bufferManager;
 
     void lookForNewLogs();
     void lookForExogenousSignals();
+
+    bool
+    addChannelAndMetadata(const std::string& nameKey, const std::vector<std::string>& metadata);
+    void storeAndSendLoggingData(const std::string& name,
+                                 const Eigen::VectorXd& data,
+                                 const double time);
+    bool addChannel(const std::string& nameKey, const std::vector<std::string>& metadata);
+    void storeLoggingData(const std::string& name,
+                                 const Eigen::VectorXd& data,
+                                 const double time);
+    bool (BipedalLocomotion::YarpRobotLoggerDevice::*initMetadataFunction)(const std::string& nameKey, const std::vector<std::string>& metadata);
+    void (BipedalLocomotion::YarpRobotLoggerDevice::*loggingDataFunction)(const std::string& name,
+                                 const Eigen::VectorXd& data,
+                                 const double time);
```

Comment:
I would avoid using function pointers as a member of the class. I would prefer two separate functions (one if realtime is enable and one if disabled)